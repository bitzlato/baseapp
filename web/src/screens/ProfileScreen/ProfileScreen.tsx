import { FC } from 'react';
import { Box } from 'web/src/components/Box/Box';
import { useDocumentTitle } from 'web/src/hooks/useDocumentTitle';
import { useT } from 'web/src/hooks/useT';
import { ProfileAccountActivity } from 'web/src/containers/ProfileAccountActivity';
import { Profile } from 'web/src/components/profile/Profile';
import { ProfileSettings } from 'web/src/components/profile/settings/ProfileSettings';
import { ApiKeys } from 'web/src/components/profile/ApiKeys';
import { Reports } from 'web/src/components/profile/reports/Reports';
import { MergeWithTelegram } from 'web/src/components/profile/mergeWithTelegram/MergeWithTelegram';

export const ProfileScreen: FC = () => {
  const t = useT();
  const title = t('page.body.profile.header.account');

  useDocumentTitle(title);

  return (
    <Box my="7">
      <Profile />
      <MergeWithTelegram />
      <ProfileSettings />
      <Reports />
      <ApiKeys />
      <ProfileAccountActivity />
    </Box>
  );
};
