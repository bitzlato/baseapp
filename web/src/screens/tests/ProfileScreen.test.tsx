import { render } from '@testing-library/react';
import { TestComponentWrapper } from 'web/src/lib/test';
import { ProfileScreen } from 'web/src/screens/ProfileScreen/ProfileScreen';

describe('ProfileScreen test', () => {
  it('should render', () => {
    const { asFragment } = render(
      <TestComponentWrapper>
        <ProfileScreen />
      </TestComponentWrapper>,
    );

    expect(asFragment().children[0]).toMatchSnapshot();
  });
});
