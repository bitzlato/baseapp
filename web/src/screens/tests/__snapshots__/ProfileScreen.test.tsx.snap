// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen test should render 1`] = `
<div
  class="my7"
>
  <div
    class="container xl"
  >
    <div
      class="cardDesktop col bodyBgColor"
    >
      <div
        class="cardHeader padding3 primaryColor"
      >
        <h4>
          Profile
        </h4>
      </div>
      <div
        class="col colSpacing2 padding3"
      >
        <div
          class="row"
        >
          <div
            class="col-12 col-md-6 mx-0"
          >
            <div
              class="pg-profile-page__box pg-profile-page__left-col__basic"
            >
              <div
                class="pg-profile-page__box-header pg-profile-page__left-col__basic__info-row"
              >
                <div
                  class="pg-profile-page__left-col__basic__info-row__block"
                >
                  <div
                    class="pg-profile-page__row pg-profile-page__details-user"
                  >
                    <p />
                  </div>
                  <div
                    class="pg-profile-page__row"
                  >
                    <h2>
                      UID: 
                    </h2>
                  </div>
                </div>
              </div>
              <div
                class="pg-profile-page__row"
              >
                <div>
                  <div
                    class="pg-profile-page__label"
                  >
                    Password
                  </div>
                  <div>
                    ************
                  </div>
                </div>
                <button
                  class="btn-block mt-3 mb-3 btn-lg btn btn-primary w-25 btn btn-primary btn-lg"
                  type="button"
                >
                  Change
                </button>
              </div>
              <div
                class="pg-profile-page__row"
              >
                <label
                  class="pg-profile-page__label"
                >
                  <div>
                    Two-factor authentication
                  </div>
                  <span
                    class="pg-profile-page__label-value__disabled"
                  >
                    Disabled
                  </span>
                </label>
                <form
                  class=""
                >
                  <div
                    class="custom-control custom-switch"
                  >
                    <input
                      class="custom-control-input"
                      id="2fa-switch"
                      type="checkbox"
                    />
                    <label
                      class="custom-control-label"
                      for="2fa-switch"
                      title=""
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-6"
          >
            <div
              class="pg-profile-page__box pg-profile-page-verification"
            >
              <h3
                class="pg-profile-page-verification__title"
              >
                Profile Verification
              </h3>
              <div
                class="pg-profile-page-verification__progress-bar"
              >
                <div
                  class="pg-profile-page-verification__progress-bar__step pg-profile-page-verification__progress-bar__step--active"
                >
                  Level
                  <span>
                     1
                  </span>
                </div>
                <div
                  class="pg-profile-page-verification__progress-bar__step pg-profile-page-verification__progress-bar__step--blocked"
                >
                  Level
                  <span>
                     2
                  </span>
                </div>
                <div
                  class="pg-profile-page-verification__progress-bar__step pg-profile-page-verification__progress-bar__step--blocked"
                >
                  Level
                  <span>
                     3
                  </span>
                </div>
                <div
                  class="pg-profile-page-verification__progress-bar__step pg-profile-page-verification__progress-bar__step--blocked"
                >
                  Level
                  <span>
                     4
                  </span>
                </div>
                <div
                  class="pg-profile-page-verification__progress-bar__step pg-profile-page-verification__progress-bar__step--blocked"
                >
                  Level
                  <span>
                     5
                  </span>
                </div>
              </div>
              <div
                class="pg-profile-page-verification__step pg-profile-page-verification__step--active"
              >
                <div
                  class="pg-profile-page-verification__step__info"
                >
                  <div
                    class="pg-profile-page-verification__step__info__title"
                  >
                    <span>
                      1. 
                    </span>
                    Email address
                  </div>
                  <div
                    class="pg-profile-page-verification__step__info__subtitle"
                  >
                    Withdrawal allowed
                  </div>
                </div>
                <div
                  class="pg-profile-page-verification__step__button pg-profile-page-verification__step__button--active"
                >
                  <a
                    href="/confirm"
                  >
                    Verify
                  </a>
                </div>
              </div>
              <div
                class="pg-profile-page-verification__step pg-profile-page-verification__step--blocked"
              >
                <div
                  class="pg-profile-page-verification__step__info"
                >
                  <div
                    class="pg-profile-page-verification__step__info__title"
                  >
                    <span>
                      2. 
                    </span>
                    Verify Phone number
                  </div>
                  <div
                    class="pg-profile-page-verification__step__info__subtitle"
                  >
                    To allow Deposits and Trades
                  </div>
                </div>
                <div
                  class="pg-profile-page-verification__step__button pg-profile-page-verification__step__button--blocked"
                >
                  <a
                    href="/confirm"
                  >
                    Verify
                  </a>
                </div>
              </div>
              <div
                class="pg-profile-page-verification__step pg-profile-page-verification__step--blocked"
              >
                <div
                  class="pg-profile-page-verification__step__info"
                >
                  <div
                    class="pg-profile-page-verification__step__info__title"
                  >
                    <span>
                      3. 
                    </span>
                    Complete your profile
                  </div>
                  <div
                    class="pg-profile-page-verification__step__info__subtitle"
                  >
                     
                  </div>
                </div>
                <div
                  class="pg-profile-page-verification__step__button pg-profile-page-verification__step__button--blocked"
                >
                  <a
                    href="/confirm"
                  >
                    Verify
                  </a>
                </div>
              </div>
              <div
                class="pg-profile-page-verification__step pg-profile-page-verification__step--blocked"
              >
                <div
                  class="pg-profile-page-verification__step__info"
                >
                  <div
                    class="pg-profile-page-verification__step__info__title"
                  >
                    <span>
                      4. 
                    </span>
                    Verify your Identity
                  </div>
                  <div
                    class="pg-profile-page-verification__step__info__subtitle"
                  >
                    Increase Withdrawal limit to 10 BTC
                  </div>
                </div>
                <div
                  class="pg-profile-page-verification__step__button pg-profile-page-verification__step__button--blocked"
                >
                  <a
                    href="/confirm"
                  >
                    Verify
                  </a>
                </div>
              </div>
              <div
                class="pg-profile-page-verification__step pg-profile-page-verification__step--blocked"
              >
                <div
                  class="pg-profile-page-verification__step__info"
                >
                  <div
                    class="pg-profile-page-verification__step__info__title"
                  >
                    <span>
                      5. 
                    </span>
                    Verify Proof of residence
                  </div>
                  <div
                    class="pg-profile-page-verification__step__info__subtitle"
                  >
                    Increase Withdrawal limit to 100 BTC
                  </div>
                </div>
                <div
                  class="pg-profile-page-verification__step__button pg-profile-page-verification__step__button--blocked"
                >
                  <a
                    href="/confirm"
                  >
                    Verify
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container xl my4"
  >
    <div
      class="cardDesktop pg-profile-page__api-keys col bodyBgColor"
    >
      <div
        class="cardHeader padding3 primaryColor"
      >
        <h4>
          My API Keys
        </h4>
      </div>
      <div
        class="col colSpacing2 padding3"
      >
        <p
          class="pg-profile-page__label pg-profile-page__text-center"
        >
          Please enable Two-factor authentication
        </p>
      </div>
    </div>
  </div>
  <div
    class="container xl my4"
  >
    <div
      class="cardDesktop pg-profile-page__activity col bodyBgColor"
    >
      <div
        class="cardHeader padding3 primaryColor"
      >
        <h4>
          Account Activity
        </h4>
      </div>
      <div
        class="col colSpacing2 padding3"
      >
        <div
          class="pg-history-elem pg-history-empty"
        />
      </div>
    </div>
  </div>
</div>
`;
